Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTClient.cs

```diff
@@ -514,36 +585,60 @@ private bool Get6DOFSettings()
             return false;
         }
 
+        // Build two skeletons in parallel.  The traditional one for use with Mecanim
+        // The other with custom, user defined, Avatars.
         private bool GetSkeletonSettings()
         {
             bool getStatus = mProtocol.GetSkeletonSettings();
             if (!getStatus)
                 return false;
 
             mSkeletons.Clear();
+            mSkeletonsCustom.Clear();
             var skeletonSettings = mProtocol.SkeletonSettingsCollection;
             foreach (var settingSkeleton in skeletonSettings.SettingSkeletonList)
             {
                 Skeleton skeleton = new Skeleton();
+                Skeleton skeletonCustom = new Skeleton();
+
                 skeleton.Name = settingSkeleton.Name;
+                skeletonCustom.Name = settingSkeleton.Name;
+
                 foreach (var settingSegment in settingSkeleton.SettingSegmentList)
                 {
                     var segment = new Segment();
+                    var segmentCustom = new Segment();
                     segment.Name = settingSegment.Name;
                     segment.Id = settingSegment.Id;
                     segment.ParentId = settingSegment.ParentId;
+                    segmentCustom.Name = settingSegment.Name;
+                    segmentCustom.Id = settingSegment.Id;
+                    segmentCustom.ParentId = settingSegment.ParentId;
 
-                    // Set rotation and position to work with unity
+
+                    // Set rotation and position to work with unity - old QAvatar
+                    // For use with Mecanim
                     segment.TPosition = new Vector3(settingSegment.Position.X / 1000, settingSegment.Position.Z / 1000, settingSegment.Position.Y / 1000);
                     segment.TRotation = new Quaternion(settingSegment.Rotation.X, settingSegment.Rotation.Z, settingSegment.Rotation.Y, -settingSegment.Rotation.W);
+                    //Debug.Log(segment.Name+" TRotation"+segment.TRotation.eulerAngles.ToString());
+
+
+                    // Set rotation and position to work with unity - User defined avatar
+                    segmentCustom.TPosition = new Vector3(-settingSegment.Position.X / 1000, settingSegment.Position.Z / 1000, settingSegment.Position.Y / 1000);
+                    segmentCustom.TRotation = new Quaternion(-settingSegment.Rotation.X, settingSegment.Rotation.Z, settingSegment.Rotation.Y, -settingSegment.Rotation.W);
```

Comment:
Is it not possible to delay these kind of adjustments until the point where we actually know how the skeleton should be handled (mecanim or not), and thus avoid having duplicate almost identical skeleton instances? Basically move this part forward to `UpdateCustom()`...

Reply:
Good question!  We considered a few different approaches but none of them challenged the idea of doing the RH to LH conversion in RTClient.cs.  Looking at RTSkeleton I don't see any reason why we couldn't push the conversion into the Update() methods there.

Reply:
@helmesjo 
This is basically leaking the conversion logic to the other scripts. The other components don't do this so it would create some confusion.

These calculations has been uniformed again so no leakage is needed.
https://github.com/qualisys/Qualisys-Unity-SDK/pull/19